loki/transform/transform_inline.py

```diff
@@ -238,17 +248,38 @@ def _map_unbound_dims(var, val):
                 if isinstance(arg, sym.Array):
                     # Resolve implicit dimension ranges of the passed value,
                     # eg. when passing a two-dimensional array `a` as `call(arg=a)`
-                    qualified_value = val if val.dimensions else val.clone(
-                        dimensions=tuple(sym.Range((None, None)) for _ in arg.shape)
-                    )
+                    # Check if val is a DeferredTypeSymbol, as it does not have a `dimensions` attribute
+                    if not isinstance(val, sym.DeferredTypeSymbol) and val.dimensions:
```

Comment:
Interesting corner case! Do you happen to have a small MFE code snippet to reproduce this? If not I'll try and add one to the main PR once I have rebased it.